src/hotspot/share/nmt/vmatree.cpp

```diff
@@ -206,7 +207,8 @@ VMATree::SummaryDiff VMATree::register_mapping(position A, position B, StateType
   }
 
   // Finally, we can register the new region [A, B)'s summary data.
-  SingleDiff& rescom = diff.tag[NMTUtil::tag_to_index(metadata.mem_tag)];
+  MemTag tag_to_change = use_tag_inplace ? stA.out.mem_tag() : metadata.mem_tag;
```

Comment:
Do we ever change the `stA.out.mem_tag()` ? Can't it always be `stA.out.mem_tag()`?

Reply:
Yes, in line 79 above.

Reply:
@afshin-zafari .

Right, but that depends on `use_tag_inplace` being true. So this says:

```c++
MemTag found_memtag; // Assume set
if (use_tag_inplace) {
  stA.mem_tag() =  found_memtag;
} else {
    stA.mem_tag() = metadata.mem_tag;
}
MemTag tag_to_change = use_tag_inplace ? stA.out.mem_tag() : metadata.mem_tag; 
```

Do you see what I'm saying? You can change the line to `MemTag tag_to_change = stA.out.mem_tag();`.

Reply:
OK. I got it.
Changed. Always `stA.out.mem_tag()` is used for accounting.